src/transaction/transaction_single_node_group_complete_manager.cpp

```diff
@@ -0,0 +1,189 @@
+		looper),
+	new single_node_group_complete_task (), "single_node_group_complete_daemon");
+  }
+
+  //
+  // final finalizes single node group commit
+  //
+  void single_node_group_complete_manager::final ()
+  {
+    if (gl_single_node_group_complete_daemon != NULL)
+      {
+	cubthread::get_manager()->destroy_daemon (gl_single_node_group_complete_daemon);
+	gl_single_node_group_complete_daemon = NULL;
+      }
+
+    delete gl_single_node_group;
+    gl_single_node_group = NULL;
+  }
+
+  //
+  // notify_log_flush notifies single node group commit.
+  //
+  void single_node_group_complete_manager::notify_log_flush_lsa (const LOG_LSA *lsa)
+  {
+    /* TO DO - disable it temporary since it is not tested */
+    return;
+
+    assert (lsa != NULL && LSA_GE (lsa, &m_latest_closed_group_log_lsa));
+
+    /* TODO - use m_latest_closed_group_stream_start_positon, m_latest_closed_group_stream_end_positon */
+    if (LSA_GT (lsa, &m_latest_closed_group_log_lsa))
+      {
+	cubthread::entry *thread_p = &cubthread::get_entry();
+	do_complete (thread_p);
+      }
+  }
+
+  //
+  // on_register_transaction - on register transaction specific to single node.
+  //
+  void single_node_group_complete_manager::on_register_transaction ()
+  {
+    /* This function is called after adding a transaction to the current group.
+     * Currently, we wakeup GC thread when first transaction is added into current group.
+     */
+    if (is_latest_closed_group_completed ()
+	&& get_current_group ().get_container ().size () == 1)
+      {
+	gl_single_node_group_complete_daemon->wakeup ();
+      }
+  }
+
+  //
+  // can_close_current_group check whether the current group can be closed.
+  //
+  bool single_node_group_complete_manager::can_close_current_group ()
+  {
+    if (!is_latest_closed_group_completed ())
+      {
+	/* Can't advance to the next group since the current group was not committed yet. */
+	return false;
+      }
+
+    if (is_current_group_empty ())
+      {
+	// no transaction, can't close the group.
+	return false;
+      }
+
+    return true;
+  }
+
+  //
+  // prepare_complete prepares group complete. Always should be called before do_complete.
+  //
+  void single_node_group_complete_manager::prepare_complete (THREAD_ENTRY *thread_p)
+  {
+    LOG_LSA closed_group_commit_lsa;
+    LOG_TDES *tdes = logtb_get_tdes (&cubthread::get_entry());
+    bool has_postpone;
+
+    if (close_current_group ())
+      {
+	tx_group &closed_group = get_latest_closed_group ();
+
+	/* TODO - Introduce parameter. For now complete group MVCC only here. Notify MVCC complete. */
+	log_Gl.mvcc_table.complete_group_mvcc (closed_group);
+	notify_group_mvcc_complete (closed_group);
+
+	log_append_group_complete (thread_p, tdes, 0, closed_group, &closed_group_commit_lsa, &has_postpone);
+	LSA_COPY (&m_latest_closed_group_log_lsa, &closed_group_commit_lsa);
+	log_wakeup_log_flush_daemon ();
```

Comment:
Just to clarify : in the future log_flush will executed on another thread (like current PR) or is executed on  single_node_group_complete_task thread ?
A TODO here will clarify.